packages/storage/src/namespace_helpers.rs

```diff
@@ -66,6 +103,35 @@ fn key_len(prefix: &[u8]) -> [u8; 2] {
     [length_bytes[6], length_bytes[7]]
 }
 
+#[cfg(feature = "iterator")]
+fn trim(namespace: &[u8], key: &[u8]) -> Result<Vec<u8>> {
+    let name_len = namespace.len();
+    if key.len() < name_len {
+        return contract_err("key shorter than namespace");
+    }
+    if &key[..name_len] != namespace {
+        return contract_err("key doesn't begin with namespace");
+    }
+    Ok(key[name_len..].to_vec())
```

Comment:
If we inline this function, we don't even need to check the two error conditions above since we iterate over `(namespace|start) ... (namespace|end)`, which means we can remove all the error handling

Reply:
I actually want to fail if the key doesn't start with namespace. Cuz that means we have some error elsewhere (better error than silently ignore). I would like to leave the check (even if it *should* never fail)

Reply:
Hmm, but then you are testing other code's behaviour here. I mean if any storage presents us any kind of wrong data we are horribly doomed as this is considered trusted. I'd rather move this into test code than polliting your implementation logic.